python/tests/test_search_algorithm.py

```diff
@@ -1,73 +1,97 @@
 from dataclasses import dataclass
-from typing import Any, Sequence
+from typing import Sequence, Mapping, Dict
 from unittest import TestCase
 
-from src.search_algorithm import CandidateClass, CandidateGenerator, \
-    CostFunction, CostMap, SearchAlgorithm
+from src.search_algorithm import (
+    CandidateClass,
+    CandidateGenerator,
+    CostMap,
+    SearchAlgorithm,
+)
 
 
 @dataclass(frozen=True)
 class TestCandidate(CandidateClass):
-    ckey: str = ""
-    cost: float = 0.0
+    name: str = ""
+    value: float = 0.0
 
     def key(self) -> str:
-        return self.ckey
+        return self.name
 
 
 class TestCandidateGenerator(CandidateGenerator[TestCandidate]):
     # pylint: disable=no-self-use
+    def __init__(self, initial_population: Sequence[float]):
+        self._initial_population = initial_population
```

Comment:
hmm should this be called `_initial_population_costs`?

Reply:
The naming isn't great, but basically, I am allowing a list of floats to be passed in as the candidates to use for the initial population, and then storing that for use by the initial population generator.

Reply:
Yeah I get why its used, just think the naming could be clearer. Also you can remove `# pylint: disable=no-self-use
` on line 23

Reply:
any suggestions for a better name?

Reply:
`_initial_population_values`?